src/main/java/nextstep/subway/line/LineService.java

```diff
@@ -0,0 +1,82 @@
+package nextstep.subway.line;
+
+import nextstep.subway.section.SectionRepository;
+import nextstep.subway.station.Station;
+import nextstep.subway.station.StationNotFoundException;
+import nextstep.subway.station.StationRepository;
+import nextstep.subway.station.StationResponse;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional
+public class LineService {
+    private final LineRepository lineRepository;
+    private final StationRepository stationRepository;
+    private final SectionRepository sectionRepository;
+
+    public LineService(LineRepository lineRepository, final StationRepository stationRepository, SectionRepository sectionRepository) {
+        this.lineRepository = lineRepository;
+        this.stationRepository = stationRepository;
+        this.sectionRepository = sectionRepository;
+    }
+
+    @Transactional(readOnly = true)
+    public List<LineResponse> findAllLines() {
+        return lineRepository.findAll()
+                .stream()
+                .map(this::createLineResponse)
+                .collect(Collectors.toList());
+    }
+
+    private LineResponse createLineResponse(Line line) {
+      return new LineResponse(
+          line.getId(),
+          line.getName(),
+          line.getColor(),
+          mapToStations(line)
+      );
+    }
+
+  private static List<StationResponse> mapToStations(final Line line) {
+      return line.getStations()
+              .stream()
+              .map(StationResponse::new)
+              .collect(Collectors.toList());
+  }
+
+  public LineResponse saveLine(LineRequest lineRequest) {
+      final Line line = lineRepository.save(new Line(lineRequest.getName(),
+          lineRequest.getColor()));
+
+      if (lineRequest.getUpStationId() != null && lineRequest.getDownStationId() != null) {
+          final Station upStation = getStationById(lineRequest.getUpStationId());
+          final Station downStation = getStationById(lineRequest.getDownStationId());
+          sectionRepository.saveAll(line.addSection(upStation, downStation, lineRequest.getDistance()));
```

Comment:
```suggestion
line.addSection(upStation, downStation, lineRequest.getDistance());
```
`cascade`옵션을 추가해주셔서 이렇게만 해도 저장이 됩니다 😄